gt/lua_lpm.c

```diff
@@ -0,0 +1,395 @@
+/*
+ * Gatekeeper - DoS protection system.
+ * Copyright (C) 2016 Digirati LTDA.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <arpa/inet.h>
+#include <lauxlib.h>
+
+#include <rte_lcore.h>
+
+#include "lua_lpm.h"
+#include "luajit-ffi-cdata.h"
+#include "gatekeeper_net.h"
+#include "gatekeeper_lpm.h"
+#include "gatekeeper_fib.h"
+
+static int
+l_str_to_prefix(lua_State *l)
+{
+	int ret;
+	struct ipaddr ip_addr;
+
+	/* First argument must be an IP prefix string. */
+	const char *prefix_str = luaL_checkstring(l, 1);
+
+	if (lua_gettop(l) != 1)
+		luaL_error(l, "Expected one argument, however it got %d arguments",
+			lua_gettop(l));
+
+	ret = parse_ip_prefix(prefix_str, &ip_addr);
+	if (ret < 0 || ip_addr.proto == ETHER_TYPE_IPv6)
```

Comment:
Replace `ip_addr.proto == ETHER_TYPE_IPv6` to `ip_addr.proto != ETHER_TYPE_IPv4`, so if we add another type in the future, the code would still be correct.

Analogous case in the IPv6 case: `l_str_to_prefix6()`.